src/Corporation/Actions.ts

```diff
@@ -242,10 +245,14 @@ export function SetSmartSupply(warehouse: Warehouse, smartSupply: boolean): void
   warehouse.smartSupplyEnabled = smartSupply;
 }
 
-export function SetSmartSupplyOption(warehouse: Warehouse, material: Material, useOption: string): void {
-  if (!corpConstants.smartSupplyUseOptions.includes(useOption)) {
-    throw new Error(`Invalid Smart Supply option '${useOption}'`);
+function assertSmartSupplyOption(value: string): asserts value is CorpSmartSupplyOption {
+  if (!corpConstants.smartSupplyUseOptions.includes(value as CorpSmartSupplyOption)) {
+    throw new Error(`Invalid Smart Supply option '${value}'`);
   }
+}
+
+export function SetSmartSupplyOption(warehouse: Warehouse, material: Material, useOption: CorpSmartSupplyOption): void {
+  assertSmartSupplyOption(useOption);
```

Comment:
If the function takes in a CorpSmartSupplyOption, we don't need to validate it again inside the function. It should either take in an unvalidated string (and do the validation here) or the validation should be done before calling this function.

Reply:
We should still validate it because players who don't use TS can call it with whatever they want.

Reply:
Players don't call ingame functions. Players only call ns functions, which actually have all parameters as `unknown`. From there we do validation on their input to narrow the types to what is actually required.

Reply:
yeah, validation is already moved there :)